### PR TITLE
feat: external kms integration

### PR DESCRIPTION
**Pull request by sheensantoscapadngan**
# Description 📣
This PR integrates external KMS endpoints to the frontend

image
image

Encryption tab in Project settings is still to be integrated in a separate effort

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/ee/services/external-kms/external-kms-service.ts [hunk 1]
```diff
@@ -51,15 +57,25 @@ export const externalKmsServiceFactory = ({
...
     const kmsSlug = slug ? slugify(slug) : slugify(alphaNumericNanoId(8).toLowerCase());
 
     let sanitizedProviderInput = "";
     switch (provider.type) {
       case KmsProviders.Aws:
         {
           const externalKms = await AwsKmsProviderFactory({ inputs: provider.inputs });
-          await externalKms.validateConnection();
+          if (provider.inputs.kmsKeyId) {
+            await externalKms.validateConnection();
```
Comment (akhilmhdh) [inline backend/src/ee/services/external-kms/external-kms-service.ts:77]: validate connection must always happen. Any reason for moving it like this?

Comment (sheensantoscapadngan) [reply backend/src/ee/services/external-kms/external-kms-service.ts:77]: Validate connection fails when kmsKeyId is not provided by the user 
image

Comment (sheensantoscapadngan) [reply backend/src/ee/services/external-kms/external-kms-service.ts:77]: we're only creating the aws key if it does not exist below the validateConnection call
image

File: backend/src/ee/routes/v1/project-router.ts [hunk 2]
```diff
@@ -171,4 +171,173 @@ export const registerProjectRouter = async (server: FastifyZodProvider) => {
...
+            id: z.string(),
+            slug: z.string(),
+            isExternal: z.boolean()
+          })
+        })
+      }
+    },
+    onRequest: verifyAuth([AuthMode.JWT]),
+    handler: async (req) => {
+      const secretManagerKmsKey = await server.services.kms.getProjectSecretManagerKmsKey(req.params.workspaceId);
```
Comment (sheensantoscapadngan) [inline backend/src/ee/routes/v1/project-router.ts:197]: remove

**Review by akhilmhdh — APPROVED** [review]
Merging and ill make changes as needed on linking with secret v2

**Pull request merged**